src/settings/DeviceSelection.tsx

```diff
@@ -42,26 +49,28 @@ export const DeviceSelection: FC<Props> = ({ devices, caption }) => {
         as="h4"
         className={styles.title}
       >
-        {caption}
+        {title}
       </Heading>
       <Separator className={styles.separator} />
       <div className={styles.options}>
-        {devices.available.map(({ deviceId, label }, index) => (
+        {[...devices.available].map(([id, label]) => (
           <InlineField
-            key={deviceId}
+            key={id}
             name={groupId}
             control={
               <RadioControl
-                checked={deviceId === devices.selectedId}
+                checked={id === devices.selectedId}
                 onChange={onChange}
-                value={deviceId}
+                value={id}
               />
             }
           >
             <Label>
-              {!!label && label.trim().length > 0
-                ? label
-                : `${caption} ${index + 1}`}
+              {label.type === "name"
+                ? label.name
+                : label.type === "number"
+                  ? numberedLabel(label.number)
+                  : t("settings.devices.default")}
```

Comment:
I think It could be nice to also include what the default device would be if selected:
![image](https://github.com/user-attachments/assets/1f950f9e-5239-4b60-b329-e39e5ee6e8c3)
Show a greyed out text behind the Default label for example: `(MacBook Pro Speakers)`
This would require
```
export type DeviceLabel =
  | { type: "name"; name: string }
  | { type: "number"; number: number }
  | { type: "default" };
```
to include the label: `{ type: "default", label: string|number};`